kbase-extension/static/kbase/js/widgets/function_output/kbaseReportView.js

```diff
@@ -233,155 +239,111 @@ define([
-                            label: _this.escapeHtml(item.label || item.name),
-                            url: [
-                                htmlServiceURL,
-                                'api',
-                                'v1',
-                                _this.objIdentity.ref,
-                                '$',
-                                index,
-                                item.name,
-                            ].join('/'),
-                            description: item.description,
-                        };
-                    });
-                } else {
-                    return [];
-                }
+
+            if (!report.html_links || !report.html_links.length) {
+                return Promise.resolve([]);
+            }
+            return serviceWizard.lookupModule().spread((info) => {
+                return report.html_links.map((item, index) => {
+                    return {
+                        name: item.name,
+                        label: StringUtil.escape(item.label || item.name),
+                        description: item.description,
+                        url: [
+                            info.url,
+                            'api',
+                            'v1',
+                            this.objIdentity.ref,
+                            '$',
+                            index,
+                            item.name,
+                        ].join('/'),
+                    };
+                });
             });
         },
 
-        makeIframeSrcUrl: function (arg) {
-            const t = html.tag,
-                iframe = t('iframe');
-
-            const iframeId = 'frame_' + html.genId();
-
-            const width = arg.width || '100%',
-                // maxHeight = arg.maxHeight || 'auto',
-                iframeHtml = iframe({
-                    style: {
-                        display: 'block',
-                        width: width,
-                        height: arg.height,
-                        margin: 0,
-                        padding: 0,
-                    },
-                    dataFrame: iframeId,
-                    frameborder: '0',
-                    scrolling: 'yes',
-                    id: iframeId,
-                    src: arg.src,
+        /**
+         * Returns a Promise that resolves into an array of created objects, or an
+         * empty array if there are none. Also, if the showCreatedObjects option isn't set,
+         * this just returns a Promise that resolves into an empty array and doesn't call out
+         * to the Workspace service.
+         * @param {object} report
+         */
+        getCreatedObjectInfo: function (report) {
+            if (
+                !this.options.showCreatedObjects || // if we're not showing it, just return an empty array
+                !report.objects_created ||
+                report.objects_created.length === 0
```

Comment:
or `!report.objects_created.length`, to make it consistent with what you did on line 289. It doesn't really matter though.

Reply:
Consistency is important! Fixed!